test/clear.js

```diff
@@ -77,3 +79,19 @@ test('incorrect clear', async function (t) {
   t.ok(await core.has(128))
   t.alike(await core.get(128), b4a.from('tick'))
 })
+
+test('clear with diff option', async function (t) {
+  const storage = createTmpDir(t)
+
+  const core = new Hypercore(storage)
+  const block = await core.append(b4a.alloc(32 * 1024).fill('a'))
+
+  console.log(block)
+  await core.clear(1)
```

Comment:
Just temporal debugging, it fails with this:

`{ length: 1, byteLength: 32768 }`

```
/self/hypercore/node_modules/fs-native-extensions/index.js:5
  if (errno < 0) this.reject(toError(errno))
                             ^

Error: EINVAL: invalid argument
    at Object.onwork (/self/hypercore/node_modules/fs-native-extensions/index.js:5:30) {
  errno: -22,
  code: 'EINVAL'
}
```